### ISSUE TITLE
Kernel process lockup on heap expansion

### ISSUE THREAD

**Issue by EdNutting** [FlingOS/FlingOS#82]
The kernel process will currently lock up or crash the whole system if the heap tries to auto expand because either; it will have run out of memory entirely, so new pages can't be added to the memory layout because the dictionary can't be expanded or; the deferred system calsl thread will get held up by the spin lock when on Alloc when trying to expand the memory layout dictionaries.

A special case needs to be implemented.
